network-store-client/src/main/java/com/powsybl/network/store/client/RestNetworkStoreClient.java

```diff
@@ -89,22 +91,43 @@ private <T extends IdentifiableAttributes> Optional<Resource<T>> get(String targ
         return resource;
     }
 
-    private <T extends IdentifiableAttributes> void updateAll(String target, String url, List<Resource<T>> resourceList, Object... uriVariables) {
-        for (List<Resource<T>> resourcePartition : Lists.partition(resourceList, RESOURCES_CREATION_CHUNK_SIZE)) {
+    private <T extends IdentifiableAttributes> void updateAll(String target, String url, AttributeFilter attributeFilter, List<Resource<T>> resources, Object[] uriVariables) {
+        if (attributeFilter == null) {
+            if (LOGGER.isInfoEnabled()) {
+                LOGGER.info("Updating {} {} resources ({})...", resources.size(), target, UriComponentsBuilder.fromUriString(url).buildAndExpand(uriVariables));
+            }
+            restClient.updateAll(url, resources, uriVariables);
+        } else {
+            List<Resource<Attributes>> filteredResources = resources.stream()
+                    .map(resource -> resource.filterAttributes(attributeFilter))
+                    .collect(Collectors.toList());
+            String filteredUrl = url + "/" + attributeFilter.name().toLowerCase();
             if (LOGGER.isInfoEnabled()) {
-                LOGGER.info("Updating {} {} resources ({})...", resourcePartition.size(), target, UriComponentsBuilder.fromUriString(url).buildAndExpand(uriVariables));
+                LOGGER.info("Updating {} {} {} resources ({})...", filteredResources.size(), target, attributeFilter, UriComponentsBuilder.fromUriString(filteredUrl).buildAndExpand(uriVariables));
             }
+            restClient.updateAll(filteredUrl, filteredResources, uriVariables);
+        }
+    }
+
+    private <T extends IdentifiableAttributes> void updateAll(String target, String url, List<Resource<T>> resources, AttributeFilter attributeFilter, Object... uriVariables) {
```

Comment:
maybe find a better name ? These two can be too easily mixed up.
```
updateAll(String target, String url, List<Resource<T>> resources, AttributeFilter attributeFilter, Object... uriVariables)
updateAll(String target, String url, AttributeFilter attributeFilter, List<Resource<T>> resources, Object[] uriVariables) 
```

Reply:
I renamed first one updateAllPartition